File: Data Structures & Algorithms/partition-equal-subset-sum/submission-5.java
class Solution {
    public boolean canPartition(int[] nums) {
        int sum = 0;
        for(int x : nums)
            sum += x;
        if(sum%2 != 0) return false;
        sum /= 2;
        return helper(nums, 0, 0, sum);
    }
    public boolean helper(int[] nums, int i, int curr, int sum) {
        if(curr==sum) {
            return true;
        }
        if(i==nums.length) return false;
        boolean x = helper(nums, i+1, curr+nums[i], sum);
        if(x) return true;
        return helper(nums, i+1, curr, sum);
    }
}

File: Data Structures & Algorithms/subsets-ii/submission-5.py
class Solution:
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        ans = []
        nums.sort()
        def helper(i: int, curr: List[int]):
            if i==len(nums):
                ans.append(curr.copy())
                return
            curr.append(nums[i])
            helper(i+1, curr)
            curr.pop()
            while i<len(nums)-1 and nums[i+1]==nums[i]:
                i += 1
            helper(i+1, curr)
        helper(0, [])
        return ans